src/simularium/VisGeometry.ts

```diff
@@ -1202,82 +1226,23 @@ class VisGeometry {
         return null;
     }
 
-    // get the json file that accompanies the trajectory
-    // this file includes geometry data
-    public mapFromJSON(
-        name: string,
-        filePath: string,
-        assetPath: string,
-        callback?: (any) => void
-    ): Promise<void | Response> {
-        if (!filePath) {
-            return Promise.resolve();
-        }
-        const jsonRequest = new Request(filePath);
-        return fetch(jsonRequest)
-            .then((response) => {
-                if (!response.ok) {
-                    return Promise.reject(response);
-                }
-                return response.json();
-            })
-            .catch((response: Response) => {
-                if (response.status === 404) {
-                    console.warn(
-                        `Could not fetch geometry info for ${name} because ${filePath} does not exist.`
-                    );
-                } else {
-                    console.error(
-                        `Could not fetch geometry info for ${name}: ${response.statusText}`
-                    );
-                }
-            })
-            .then((data) => {
-                if (data) {
-                    this.setGeometryData(
-                        data as AgentTypeVisDataMap,
-                        assetPath,
-                        callback
-                    );
-                } else {
-                    this.resetMapping();
-
-                    // if there is an id to color mapping, set up with spheres
-                    for (const i of this.idColorMapping.keys()) {
-                        this.visGeomMap.set(i, {
-                            meshName: DEFAULT_MESH_NAME,
-                            pdbName: "",
-                        });
-                    }
-                }
-            });
+    public handleAgentGeometry(typeMapping: EncodedTypeMapping): void {
+        this.clearForNewTrajectory();
+        this.setGeometryData(typeMapping);
     }
 
-    private setGeometryData(
-        jsonData: AgentTypeVisDataMap,
-        assetPath: string,
-        callback?: (any) => void
-    ): void {
-        // clear things out in advance of loading all new geometry
-        this.resetMapping();
-
-        this.logger.debug("JSON Mesh mapping loaded: ", jsonData);
-
-        Object.keys(jsonData).forEach((id) => {
-            const entry: AgentTypeVisData = jsonData[id];
-            if (id === "size") {
-                console.log("WARNING: Ignoring deprecated bounding box data");
-            } else {
-                // mesh name is entry.mesh
-                // pdb name is entry.pdb
-                this.mapIdToGeom(Number(id), entry.mesh, entry.pdb, assetPath);
-                this.setScaleForId(Number(id), entry.scale);
```

Comment:
did we lose the `scale` value?

Reply:
Yeah, I think it's been gone since version 2? I'm going based on the doc. @blairlyons ?